enhancements/api-review/add-ImagePolicy-to-config.openshift.io-v1alpha1.md

```diff
@@ -0,0 +1,391 @@
+	// +optional
+	Repository string `json:"repository,omitempty"`
+	// remapPrefix is the prefix of the image identity to be matched. This field is required if identityMatchPolicy is set to "remapIdentity".
+	// +optional
+	RemapPrefix string `json:"remapPrefix,omitempty"`
+	// remapSignedPrefix is the prefix of the image identity to be matched in the signature. This field is required if identityMatchPolicy is set to "remapIdentity".
+	// +optional
+	RemapSignedPrefix string `json:"remapSignedPrefix,omitempty"`
+}
+
+// IdentityMatchPolicy defines the type of matching for setting "matchPolicy".
+// Valid values are "MatchRepoDigestOrExact", "MatchRepository", "ExactRepository", "RemapIdentity". When omitted, the default value is "MatchRepoDigestOrExact".
+type IdentityMatchPolicy string
+
+const (
+	// "MatchRepoDigestOrExact" means that the identity in the signature must be in the same repository as the image identity if the image identity is referenced by a digest. Otherwise, the identity in the signature must be the same as the image identity.
+	IdentityMatchPolicyMatchRepoDigestOrExact IdentityMatchPolicy = "MatchRepoDigestOrExact"
+	// "MatchRepository" means that the identity in the signature must be in the same repository as the image identity.
+	IdentityMatchPolicyMatchRepository IdentityMatchPolicy = "MatchRepository"
+	// "ExactRepository" means that the identity in the signature must be in the same repository as a specific identity specified by "dockerRepository".
+	IdentityMatchPolicyExactRepository IdentityMatchPolicy = "ExactRepository"
+	// "remapIdentity" means that the signature must be in the same as the remapped image identity.
+	IdentityMatchPolicyRemapIdentity IdentityMatchPolicy = "RemapIdentity"
+)
+
+// +k8s:deepcopy-gen=true
+type ImagePolicyStatus struct {
+
+	// policyJSON contains the whole policy applied to the namespace
+	// which got written to disk. This includes cluster-wide policies
+	// from the `openshift-config` namespace as well.
+	PolicyJSON string `json:"policyJSON,omitempty"`
+
+	// conditions provide details on the status of the gatherer job.
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	Conditions []metav1.Condition `json:"conditions" patchStrategy:"merge" patchMergeKey:"type"`
+}
+```
+
+### Implementation Details/Notes/Constraints [optional]
+
+#### Update container runtime config controller to watch ImagePolicy
+
+MCO container runtime config controller ImagePolicy syncHandler updates the signature verification configurations:
+- retrieves all the ImagePolicy instances on the cluster.
+- validates conflicting data from ImagePolicy instances. Rejects to roll out the CRs and reports the error to the machine config controller logs if the sigstore policy for the same image scope has been configured under global(openshift-config) namespace and non global namespace CRs. 
```

Comment:
The “merge rules” section says that the global settings are used and the non-global ones are ignored (with the conflict logged in `Status`).

The two texts should probably be consistent. (I prefer the variant of ignoring the non-global ones, that way a namespace admin can’t sabotage the global config.)

Reply:
fixed as the merge rules, ignore the non-global ones.